File: src/demo_programs/turtle-snowflake.py
# 3066879e89a244ba9008af1e60ca8542fb45ae648570f7f286b282c76073fcf8 Elan 1.9.0 guest default_profile valid

def main -> None:  # [main]
  t = new Turtle() # [variable definition]
  t.placeAt(-50, 30) # [call procedure]}
  t.turn(90) # [call procedure]}
  for i in range(1, 3, 1):  # [for loop]
    drawSide(side, t) # [call procedure]}
    t.turn(120) # [call procedure]}


def drawSide(length: Float, t: Turtle) -> None:  # [procedure]
  if (length > 1) then
    third = length/3 # [variable definition]
    drawSide(third, t) # [call procedure]}
    t.turn(-60) # [call procedure]}
    drawSide(third, t) # [call procedure]}
    t.turn(120) # [call procedure]}
    drawSide(third, t) # [call procedure]}
    t.turn(-60) # [call procedure]}
    drawSide(third, t) # [call procedure]}
  else
    t.move(length) # [call procedure]}


side = 100

File: test/files/turtle-snowflake.java
// Java with Elan 2.0.0-beta2

public class Global {

static void main() {
  var t = new Turtle();
  t.placeAt(-50, 30); // procedure call
  t.turn(90); // procedure call
  foreach (var i in range(1, 4)) {
    drawSide(side, t); // procedure call
    t.turn(120); // procedure call
  } // end foreach
} // end main

static void drawSide(double length, Turtle t) { // procedure
  if ((length > 1)) {
    var third = length/3;
    drawSide(third, t); // procedure call
    t.turn(-60); // procedure call
    drawSide(third, t); // procedure call
    t.turn(120); // procedure call
    drawSide(third, t); // procedure call
    t.turn(-60); // procedure call
    drawSide(third, t); // procedure call
  } else {
    t.move(length); // procedure call
  } // end if
} // end procedure

static final Int side = 100; // constant
} // end Global
